File: iOS/TUILiveKit/Sources/Component/Gift/Model/TUILikeData.swift
//
//  TUILikeData.swift
//  TUILiveKit
//
//  Created by krabyu on 2024/4/2.
//

import Foundation
import RTCRoomEngine

public class TUILikeData {
    public var sender: TUIUserInfo = TUIUserInfo()

    public init() {}
    
    public init(sender: TUIUserInfo) {
        self.sender = sender
    }
}

File: Flutter/live_uikit_gift/lib/state/tui_like_data.dart
import 'package:rtc_room_engine/rtc_room_engine.dart';

class TUILikeData {
  TUIUserInfo sender = TUIUserInfo(
      userId: '', userName: '', avatarUrl: '', userRole: TUIRole.generalUser);

  TUILikeData({TUIUserInfo? sender}) {
    if (sender != null) {
      this.sender = sender;
    }
  }

  @override
  String toString() {
    return 'TUILikeData{sender: $sender}';
  }
}